compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -572,6 +572,9 @@ error.invalid.listener.attachment=\
 error.invalid.action.invocation.syntax=\
   invalid remote function invocation syntax, use ''->'' operator
 
+error.invalid.init.invocation=\
+  call to function ''__init'' is not allowed here
```

Comment:
Should we add a more descriptive error message?

Reply:
```suggestion
  object ''__init()'' cannot be called outside the object
```

Reply:
@pubudu91 thoughts?

Reply:
+1 for a more descriptive error message. Maybe something like below:
```
call to function ''__init'' is only allowed within its object type descriptor
```

Reply:
Changed the message to `
''__init'' function invocation is only allowed within its object type descriptor`

Reply:
With other error messages related to init we seem to have used ''__init()''. Shall we keep this uniform?

Also saying "its" doesn't feel right here, unless we start with "object ''__init()''..." This should also be "method" and not "function".

How about
```suggestion
  object ''__init()'' method invocation is only allowed within the type descriptor
```

@irshadnilam, let's finalize the error message before doing any more changes :)

Reply:
As discussed offline let's change this to 
```suggestion
  object ''__init'' method invocation is only allowed within the type descriptor
```

Shall we also please change
```
error.object.init.function.cannot.be.extern=\
  object ''__init()'' function cannot be an ''external'' function
```

to 

```
error.object.init.function.cannot.be.extern=\
  object ''__init'' method cannot be an ''external'' function
```

Reply:
Changed.